internal/controllers/machine/drain/filters.go

```diff
@@ -117,48 +131,43 @@ const (
 // MakePodDeleteStatusOkay is a helper method to return the corresponding PodDeleteStatus.
 func MakePodDeleteStatusOkay() PodDeleteStatus {
 	return PodDeleteStatus{
-		Delete: true,
-		Reason: PodDeleteStatusTypeOkay,
+		DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+		DrainOrder:    ptr.To[int32](0),
+		Reason:        PodDeleteStatusTypeOkay,
 	}
 }
 
 // MakePodDeleteStatusSkip is a helper method to return the corresponding PodDeleteStatus.
 func MakePodDeleteStatusSkip() PodDeleteStatus {
 	return PodDeleteStatus{
-		Delete: false,
-		Reason: PodDeleteStatusTypeSkip,
+		DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorSkip,
+		Reason:        PodDeleteStatusTypeSkip,
 	}
 }
 
 // MakePodDeleteStatusWithWarning is a helper method to return the corresponding PodDeleteStatus.
-func MakePodDeleteStatusWithWarning(del bool, message string) PodDeleteStatus {
+func MakePodDeleteStatusWithWarning(behavior clusterv1.MachineDrainRuleDrainBehavior, message string) PodDeleteStatus {
+	var order *int32
+	if behavior == clusterv1.MachineDrainRuleDrainBehaviorDrain {
+		order = ptr.To[int32](0)
```

Comment:
MakePodDeleteStatusOkayWithOrder()?

Reply:
Compared to MakePodDeleteStatusOkay this func also takes a message.

In general I tried not to change too much to keep the filters from kubectl drain diff'able

Reply:
I added a MakePodDeleteStatusOkayWithOrder func, but I'm only using it in `machineDrainRulesFilter`